TypeCobol/Compiler/Preprocessor/ReplacingTokensLinesIterator.cs

```diff
@@ -22,12 +24,43 @@ public ReplacingTokensLinesIterator(ITokensLinesIterator sourceIterator, CopyDir
             _copyReplacingDirective = copyReplacingDirective;
         }
 
-        protected override CheckTokenStatus CheckNextTokenBeforeReplace(IReadOnlyList<ReplaceOperation> currentReplaceOperations)
+        protected override CheckTokenStatus CheckNextTokenBeforeReplace([CanBeNull] IReadOnlyList<ReplaceOperation> currentReplaceOperations)
         {
             var nextToken = SourceIteratorNextToken();
 
-            // TODO: Add error to signal unsupported feature or implement
-            // This parser currently does not support REPLACE coming from a copy and that should be altered by the REPLACING clause
+            // This parser currently does not support REPLACE that are altered by the REPLACING clause
+            if (nextToken.TokenType == TokenType.REPLACE_DIRECTIVE)
+            {
+                //TODO ReplaceAndReplacing better code to find the real ReplaceDirective on the line. Property ReplaceDirective is the LAST on the line
```

Comment:
The ReplaceIterator casts the token into `CompilerDirectiveToken` to retrieve the directive but your code should work too.